script/app.js

```diff
@@ -58,28 +60,108 @@ class MyBook {
   };
 }
 
-document.addEventListener('DOMContentLoaded', () => {
-  const newBook = document.querySelector('#newBook');
-  newBook.addEventListener('submit', (event) => {
-    event.preventDefault();
-    const title = event.target.elements[0].value;
-    const author = event.target.elements[1].value;
-    const myNewBook = new MyBook(title, author);
-    myNewBook.addBookToDom(); // append book to the DOM
-    event.target.elements[0].value = '';
-    event.target.elements[1].value = '';
+const updateBookList = () => {
+  const bookData = JSON.parse(localStorage.getItem('bookCollection'));
+  bookData.forEach((bookData) => {
+    const newBook = new MyBook(bookData.title, bookData.author, bookData.id);
+    newBook.addBookToDom(); // append book to the DOM
   });
+};
+
+const addNewBookEvent = () => {
+  try {
+    const newBook = document.querySelector('#newBook');
+    newBook.addEventListener('submit', (event) => {
+      event.preventDefault();
+      const title = event.target.elements[0].value;
+      const author = event.target.elements[1].value;
+      const myNewBook = new MyBook(title, author);
+      MyBook.unRender.push(myNewBook);
+      event.target.elements[0].value = '';
+      event.target.elements[1].value = '';
+    });
+  } catch (e) {
+    console.log(e);
+  }
+};
+document.addEventListener('DOMContentLoaded', () => {
   if (!localStorage.getItem('bookCollection')) {
     try {
       localStorage.setItem('bookCollection', JSON.stringify([]));
     } catch (exc) {
       console.log('failed to create local storage');
     }
   } else {
-    const bookData = JSON.parse(localStorage.getItem('bookCollection'));
-    bookData.forEach((bookData) => {
-      const newBook = new MyBook(bookData.title, bookData.author);
-      newBook.addBookToDom(); // append book to the DOM
+    updateBookList();
+  }
+});
+
+let body = null;
+let footer = null;
+let nav = null;
+let addBook = null;
+let ListOfBooks = null;
+let contactInfo = null;
+const swapSection = (section) => {
+  contactInfo.remove();
+  ListOfBooks.remove();
+  addBook.remove();
+  body.insertBefore(section, footer);
+  if (section === addBook) {
+    addNewBookEvent();
+  } else if (section === ListOfBooks && MyBook.unRender.length > 0) {
+    MyBook.unRender.forEach((book) => {
+      book.addBookToDom();
```

Comment:
- [ ] Every time I add a book, the first one adds properly, but it continues to loop and print empty strings. This is what I mean:

![image](https://user-images.githubusercontent.com/77106962/158701536-2eb3d59d-4cbb-4dc4-acdb-35e41f44c2c7.png)

I would advise you to render once or use `classList.add` or `classList.remove` method to switch pages

Reply:
@Gambit142  Thank you, we will make the changes before requesting a new review.

Reply:
Dear @Gambit142 we have fixed the duplicate title and author issue in the commit hash https://github.com/Tadesse-Alemayehu/Awesome-books/pull/3/commits/e288a62dcb74dafb04ffafef4f414683db7f9beb and removed all console.log statements in https://github.com/Tadesse-Alemayehu/Awesome-books/pull/3/commits/0a9a3f8d7c1052f8a99c056eb717ae2994a7f8d4 commit hash.

Reply:
Dear @Tadesse-Alemayehu @oshanedesign , I am glad you fixed the bugs and implemented the required changes. You are on the right path and I have no doubt you will be a great developers because being open to feedback helps us become better developers 👍

Reply:
@Gambit142 It's a relief to know we are on the right track. Thank you.